cellprofiler/utilities/version.py

```diff
@@ -92,10 +92,23 @@ def get_version():
         import cellprofiler.frozen_version
         return cellprofiler.frozen_version.version_string
 
+def get_dotted_version():
```

Comment:
I don’t know if this is the appropriate place to discuss, but we should consider rethinking our existing versioning process. A few observations:

-	a version number (e.g. 2.2.0) determines CellProfiler’s most recent release
-	a version doesn’t describe the release channel (e.g. nightly, beta, or stable)
-	a version (i.e. frozen_version) includes the object information generated by Git (e.g. a commit’s SHA-1)

A few opinions about versioning from a debugging perspective:

-	a version number is useful for determining when issues arise or resolve
-	a version channel is useful for narrowing when an issues arise or resolve
-	object information is useful for pinpointing when an issue arose or resolved

However, object information is only useful if the debugger is capable of running arbitrary versions, channel distributions, and commits. While it’s certainly possible, it’s difficult and usually impractical. Moreover, I think testing every commit on Travis CI mostly addresses this.

I don’t know. On one hand, a SHA-1 is undoubtedly useful when debugging. On the other hand, it requires a lot of baggage.